packages/redirect-parser/src/merge.ts

```diff
@@ -10,14 +8,18 @@ import { splitResults } from './results.js'
 export const mergeRedirects = function ({ fileRedirects, configRedirects }) {
   const results = [...fileRedirects, ...configRedirects]
   const { redirects, errors } = splitResults(results)
-  const mergedRedirects = redirects.filter(isUniqueRedirect)
+  const mergedRedirects = removeDuplicates(redirects)
   return { redirects: mergedRedirects, errors }
 }
 
 // Remove duplicates. This is especially likely considering `fileRedirects`
 // might have been previously merged to `configRedirects`, which happens when
 // `netlifyConfig.redirects` is modified by plugins.
-// The latest duplicate value is the one kept.
-const isUniqueRedirect = function (redirect, index, redirects) {
-  return !redirects.slice(index + 1).some((otherRedirect) => isDeepStrictEqual(redirect, otherRedirect))
+// The latest duplicate value is the one kept, hence why we need to reverse the
+// arrays in order to keep said logic.
```

Comment:
If the lines are identical, why does it matter which one is kept?

Reply:
The lines are identical however there might be other redirects with the same `from` property, changing the order of the duplicate we keep now could effectively tamper with that logic. A practical example, given the non-deduped array:

```js
[
  {
    from: "this-is-not-unique"
    to: "this-too-is-not-unique"
  },
  {
    from: "this-is-not-unique"
    to: "this-IS-UNIQUE"
  },
  {
    from: "this-is-not-unique"
    to: "this-too-is-not-unique"
  },
]
```

Assuming we wouldn't keep said logic and effectively keep the first duplicate, this would render:

```js
[
  {
    from: "this-is-not-unique"
    to: "this-too-is-not-unique"
  },
  {
    from: "this-is-not-unique"
    to: "this-IS-UNIQUE"
  },
]
```

Now, according to our docs - https://docs.netlify.com/routing/redirects/#rule-processing-order -  the end result for this case given that:
>  "The redirects engine will process the first matching rule it finds, reading from top to bottom. Rules in the _redirects file are always processed first, followed by rules in the Netlify configuration file."
The redirect that would end up being deployed would be the first one `to` `"this-too-is-not-unique"`.

The current implementation however would render a different scenario, `to` `"this-IS-UNIQUE"` would be deployed.

So tl;dr changing that behaviour would be a breaking change.